romancal/skycell/skymap.py

```diff
@@ -0,0 +1,724 @@
+                inside=self.vectorpoint_center,
+            )
+
+    @property
+    def pixel_scale(self) -> float:
+        """degrees per pixel"""
+        return SKYMAP.pixel_scale
+
+    def __eq__(self, other) -> bool:
+        if not isinstance(other, ProjectionRegion):
+            return NotImplemented
+
+        return self.data == other.data
+
+
+def wcsinfo_to_wcs(
+    wcsinfo: dict | stnode.Wcsinfo,
+    bounding_box: None | tuple[tuple[float, float], tuple[float, float]] = None,
+) -> WCS:
+    """Create a WCS from the L3 wcsinfo meta
+
+    Parameters
+    ----------
+    wcsinfo : dict or MosaicModel.meta.wcsinfo
+        The L3 wcsinfo to create a WCS from.
+
+    bounding_box : None or 4-tuple
+        The bounding box in detector/pixel space. Form of input is:
+        (x_left, x_right, y_bottom, y_top)
+
+    Returns
+    -------
+    wcs : WCS
+        The WCS object created.
+    """
+    pixelshift = models.Shift(-wcsinfo["x0_projection"], name="crpix1") & models.Shift(
+        -wcsinfo["y0_projection"], name="crpix2"
+    )
+    pixelscale = models.Scale(wcsinfo["pixel_scale"], name="cdelt1") & models.Scale(
+        wcsinfo["pixel_scale"], name="cdelt2"
+    )
+    tangent_projection = models.Pix2Sky_TAN()
+    celestial_rotation = models.RotateNative2Celestial(
+        wcsinfo["ra_projection_center"], wcsinfo["dec_projection_center"], 180.0
+    )
+
+    matrix = wcsinfo.get("rotation_matrix", None)
+    if matrix is not None:
+        matrix = np.array(matrix)
+    else:
+        matrix = np.reshape(
+            wcs_util.calc_rotation_matrix(
+                np.deg2rad(wcsinfo.get("orientat_projection_center", 0.0)),
+                v3i_yangle=0.0,
+                vparity=1,
+            ),
+            (2, 2),
+        )
+    rotation = models.AffineTransformation2D(matrix, name="pc_rotation_matrix")
+    det2sky = (
+        pixelshift | rotation | pixelscale | tangent_projection | celestial_rotation
+    )
+    det2sky.name = "linear_transform"
+
+    detector_frame = coordinate_frames.Frame2D(
+        name="detector", axes_names=("x", "y"), unit=(u.pix, u.pix)
+    )
+    sky_frame = coordinate_frames.CelestialFrame(
+        reference_frame=coordinates.ICRS(), name="icrs", unit=(u.deg, u.deg)
+    )
+    wcsobj = WCS(
+        [(detector_frame, det2sky), (sky_frame, None)], name=wcsinfo.get("name", None)
+    )
+
+    if bounding_box:
+        wcsobj.bounding_box = bounding_box
+
+    return wcsobj
+
+
+def skycell_to_wcs(skycell_record: dict) -> WCS:
+    """From a skycell record, generate a WCS
+
+    Parameters
+    ----------
+    skycell_record : dict
+        A skycell record, or row, from the skycell patches table.
+
+    Returns
+    -------
+    wcsobj : WCS
+        The WCS object from the skycell record.
+    """
+    wcsinfo = dict()
+
+    # The scale is given in arcseconds per pixel. Convert to degrees.
+    wcsinfo["pixel_scale"] = float(skycell_record["pixel_scale"]) / 3600.0
```

Comment:
This is not covered by tests and currently fails. It should be `skycell_record.pixel_scale`
`

Reply:
Similar access to the fields below.
It's best to add a test.